### PR TITLE
fix: Remove use of cbor, simplify exposing lalrpop in testing

### PR DESCRIPTION
**Pull request by dbcfd**

### CODE REVIEW

File: recon/src/recon/tests.rs [hunk 1]
```diff
@@ -416,378 +418,379 @@ fn hello() {
...
-        -> (a, 0, b, 0, c, 0, g) [a,b,c,e,f,g]
-        <- (a, h(b), c, 0, e, 0, f, 0, g) [a,b,c,e,f,g]
-        "#;
+    cat: [a,b,c]
+    dog: [e,f,g]
+    -> (a, h(b), c) [a,c,e,f,g]
+    <- (a, 0, c, h(e,f), g) [a,b,c,g]
+    -> (a, 0, b, 0, c, 0, g) [a,b,c,e,f,g]
+    <- (a, h(b), c, 0, e, 0, f, 0, g) [a,b,c,e,f,g]
+    "#;
```
Comment (nathanielc) [inline recon/src/recon/tests.rs:492]: why have all these test cases changed their whitespace?

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -17,6 +17,12 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
+    - name: Free Disk Space (Ubuntu)
+      uses: jlumbroso/free-disk-space@main
+      with:
+        # this might remove tools that are actually needed,
+        # when set to "true" but frees about 6 GB
+        tool-cache: true
```
Comment (nathanielc) [inline .github/workflows/rust.yml:25]: why was this needed?

File: Cargo.toml [hunk 3]
```diff
@@ -31,18 +32,19 @@ futures = "0.3"
...
+iroh-rpc-types = { git = "https://github.com/nathanielc/beetle", branch = "custom-behaviour" }
+iroh-store = { git = "https://github.com/nathanielc/beetle", branch = "custom-behaviour" }
+iroh-util = { git = "https://github.com/nathanielc/beetle", branch = "custom-behaviour" }
 lazy_static = "1.4"
 libipld = "0.15"                                                                          # use same version as Iroh
 libp2p = { version = "0.51", default-features = false }                                   # use same version as Iroh
 libp2p-identity = { version = "0.1.2", features = ["ed25519"] }
 libp2p-tls = { version = "0.1.0", default-features = false }                              # use explicit version of dep to avoid conflict
 lru = "0.8"
+minicbor = { version = "0.19.1", features = ["alloc"] }
```
Comment (nathanielc) [inline Cargo.toml:47]: Curious why the choice of minicbor? I have a WIP PR #61 where I expand our usage of cbor. Still basic usage. I can update it after this PR merges, but I'd like to understand what the other options where and why this one was selected.

**Review by nathanielc — DISMISSED** [review]
Thanks for the clean up, some questions below.

File: .github/workflows/rust.yml (hunk 2 above)
Comment (nathanielc) [reply .github/workflows/rust.yml:25]: Also looks like clang might have been removed by this step?

File: recon/src/recon/tests.rs (hunk 1 above)
Comment (dbcfd) [reply recon/src/recon/tests.rs:492]: This was the result of `cargo fmt --all`. Maybe originally formatted with an older fmt?

Comment (nathanielc) [reply recon/src/recon/tests.rs:492]: CI runs `cargo fmt --all -- --check` to ensure everything is formatted and runs on latest Rust release. Wonder why it changed.

Comment (AaronGoldman) [reply recon/src/recon/tests.rs:492]: I didn't know cargo fmt messed with`r#""#` string literals.

Comment (dbcfd) [reply recon/src/recon/tests.rs]: Figured it out, some of it was fmt, some of it was IDE auto formatting.

**Review by AaronGoldman — APPROVED** [review]
